#2917 Added Point locator test fields validation in Modbus IP

Patrykb0802 opened the pull request:
Added integer/byte validation for fields:
 - slave ID (int)
 - offset (int)
 - bit (byte)
 - register count (int)

inside point locator test for modbus ip data source.

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

Added integer/byte validation for fields:
 - slave ID (int)
 - offset (int)
 - bit (byte)
 - register count (int)

inside point locator test for modbus ip data source.

Limraj commented on WebContent/resources/common.js:
```diff
@@ -1123,3 +1123,14 @@ function trim(value) {
+    let num = Number(value);
+
+    if (!isNaN(num) && Number.isInteger(num)) {
+        if (num >= -128 && num <= 127) {
```
Can this be a negative value?
You always try to follow what already exists, look at how it works and what the isInt32 function checks. You're not reinventing the wheel.

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -67,18 +67,35 @@
+      }  else if(!isValid($get("test_bit"))) {
+            let bitMessage = createValidationMessage("test_bit", "<fmt:message key='badIntegerFormat'/>");
+            showDwrMessages([bitMessage]);
+      }  else if(!isByte($get("test_bit"))) {
```
If the value is a byte, it also meets the isValid condition. 
In the function that checks isByte, you can first check whether isValid is present and then add a limit on the value.

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Small tweaks to isByte function

Limraj commented on WebContent/resources/common.js:
```diff
@@ -1123,3 +1123,8 @@ function trim(value) {
 }
 
+function isByte(value) {
+    let num = Number(value);
```
What happens if an alphanumeric character enters here?

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -67,18 +67,32 @@
+            let registerCountMessage = createValidationMessage("test_registerCount", "<fmt:message key='badIntegerFormat'/>");
+            showDwrMessages([registerCountMessage]);
+      } else {
+          var locator = {};
```
let :)

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- changed how variables are stored inside locatorTest() function
- Added createAndDisplayValidationMessage function for showing validation message on chosen node

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Added catching exception for invalid end offset for Modbus IP data source

Limraj commented on src/com/serotonin/mango/web/dwr/DataSourceEditDwr.java:
```diff
@@ -532,7 +532,10 @@ private void testModbusPointLocator(ModbusMaster modbusMaster,
             response.addMessage(new LocalizableMessage(
                     "validate.invalidCharset"));
-        } finally {
+        } catch (ModbusIdException e) {
```
Exception

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -68,17 +68,37 @@
+          let locator = {};
+          locator.slaveId = parseInt(slaveId);
+          locator.range = range
+          locator.modbusDataType = modbusDataType;
```
Aren't modbusDataType and range numeric types?

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Minor changes

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Added Modbus IP properties fields validation for integers
- Changed validation mechanism

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Added validateValue() function for shorter code

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,20 +65,80 @@
-      
-      locatorTestImpl(locator);
+
+      let temp =  createTempLocator();
```
createLocatorConfigTemp

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,20 +65,80 @@
-      locatorTestImpl(locator);
+
+      let temp =  createTempLocator();
+      let tempModbusData = createTempModbusData();
```
createModbusConfigTemp

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbusIp.jsp:
```diff
@@ -55,13 +55,95 @@
+
+  function createTempModbusData(){
+
+    let dataSourceName = $get("dataSourceName");
```
There is no point in creating additional variables here, you get and immediately set them in the temp structure, you don't parse here, etc.

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbusIp.jsp:
```diff
@@ -55,13 +55,95 @@
+    let messages = validateModbusProperties(temp);
+
+    if(messages.length > 0) {
+      showDwrMessages(messages);
```
The formatting needs to be improved

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,20 +65,80 @@
+      hideContextualMessages("locatorTestDiv");
+      hideContextualMessages("dataSourceProperties");
+
+      let messagesModbus = validateModbusProperties(tempModbusData);
```
modbusMessages, validateModbusConfig

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,20 +65,80 @@
       setDisabled("scanCancelBtn", !scanning);
   }
+
+  function validateLocatorTest(temp){
```
temp -> locator

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbusIp.jsp:
```diff
@@ -55,13 +55,95 @@
+    let encapsulated = $get("encapsulated");
+    let createSocketMonitorPoint = $get("createSocketMonitorPoint");
+
+    let temp = {};
```
temp -> modbus

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Variables names changes

Patrykb0802 committed:
#2845 Fixed message validation in event search

- Fixed formatting

Limraj committed:
Merge branch 'release/2.7.8' into feature/#2917_Added_Point_locator_test_fields_validation_in_Modbus_IP

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,19 +65,80 @@
+  }
+
+  function createLocatorConfigTemp(){
+      let slaveId = $get("test_slaveId");
```
Similarly to the previously submitted suggestion, this is unnecessary. 
You should apply a given note to all changes you make so that you don't have to point out the same things one after the other.

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Optimized createLocatorConfigTemp() code

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,19 +65,72 @@
-      locator.charset = $get("test_charset");
-      
-      locatorTestImpl(locator);
+      setDisabled("locatorTestBtn", true);
```
You need to remove this line:

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbusIp.jsp:
```diff
@@ -56,13 +56,76 @@
+      let messages = validateModbusConfig(temp);
+
+      if(messages.length > 0) {
+		  showDwrMessages(messages);
```
stopImageFader("dsSaveImg");

Limraj commented on WebContent/WEB-INF/jsp/dataSourceEdit/editModbus.jsp:
```diff
@@ -65,19 +65,71 @@
+      let messages = modbusMessages.concat(messagesLocator);
+
+      if(messages.length > 0) {
+          showDwrMessages(messages);
```
stopImageFader("dsSaveImg");

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Added stopImageFader("dsSaveImg")

Patrykb0802 committed:
#2917 Added Point locator test fields validation in Modbus IP

- Added validation for Modbus read data table
